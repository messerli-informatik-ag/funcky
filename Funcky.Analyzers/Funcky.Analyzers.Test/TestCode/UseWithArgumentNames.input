using System;
using System.Linq;
using Funcky.CodeAnalysis;

namespace ConsoleApplication1
{
    class Program
    {
        private void Syntax()
        {
            Method(x: 10, 20);
            Method(10, 20);
            Method(
                10, 20);
            Method(
                10,
                20);
            MethodWithKeywordAsArgument(10);
        }

        [UseWithArgumentNames]
        private void Method(int x, int y) { }

        [UseWithArgumentNames]
        private void MethodWithKeywordAsArgument(int @int) { }
    }
}

namespace Funcky.CodeAnalysis
{
    [AttributeUsage(AttributeTargets.Method)]
    internal sealed class UseWithArgumentNamesAttribute : Attribute { }
}
