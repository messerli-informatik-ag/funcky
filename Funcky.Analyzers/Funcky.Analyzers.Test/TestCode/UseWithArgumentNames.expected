using System;
using System.Linq;
using Funcky.CodeAnalysis;

namespace ConsoleApplication1
{
    class Program
    {
        private void Syntax()
        {
            Method(x: 10, y: 20);
            Method(x: 10, y: 20);
            Method(
                x: 10, y: 20);
            Method(
                x: 10,
                y: 20);
            MethodWithKeywordAsArgument(@int: 10);
        }

        [UseWithArgumentNames]
        private void Method(int x, int y) { }

        [UseWithArgumentNames]
        private void MethodWithKeywordAsArgument(int @int) { }
    }
}

namespace Funcky.CodeAnalysis
{
    [AttributeUsage(AttributeTargets.Method)]
    internal sealed class UseWithArgumentNamesAttribute : Attribute { }
}
